apisix/core/log.lua

```diff
@@ -99,7 +102,10 @@ setmetatable(_M, {__index = function(self, cmd)
 
     -- cache the lazily generated method in our
     -- module table
-    _M[cmd] = method
+    if ngx.get_phase() ~= "init" then
```

Comment:
we can add `local get_phase = ngx.get_phase` at header of file, and use `get_phase()`